docs/spec/spec.md

```diff
@@ -1459,10 +1468,166 @@ parseHeader(string headerValue) returns HeaderValue[]|ClientError  {
+whatever below the next() method will only get executed after the completion of execution of the next Interceptor.
+
+If the users forget to call next() and don't respond either, the request will be left hanging.
+
+##### 8.1.1.3 Returning error?
+Resource functions can return only error?. In the case of an error, interceptor chain execution jumps to the last 
+interceptor in the chain. The last interceptor in the chain must be a type of RequestErrorInterceptor. It is a special kind of 
+interceptor which could be used to handle errors. More on this can be found under section RequestErrorInterceptor.
+
 #### 8.1.2 Request error interceptor
-// To be added
+As mentioned above this is a special kind of interceptor designed to handle errors. These interceptors can only be at the end 
+of request or response interceptor chain. Meaning these interceptors are only executed as the last interceptor in the chain. The 
+framework automatically adds default RequestErrorInterceptor and ResponseErrorInterceptor which basically prints the error 
+message to the console.
+
+Users can override these interceptors by defining their own ones as follows. Users don’t have to specifically engage 
+these interceptors as they only have fixed positions and they are always executed. The only additional argument in this 
+case is error err.
+
+```ballerina
+service class RequestErrorInterceptor {
+   *http:RequestErrorInterceptor;
+ 
+   remote function 'default [string… path](http:FilterContext ctx, http:Caller caller,
+                       http:Request req, error err) returns error? {
+       // deal with the error
+   }
+}
+```
+
+#### 8.1.3 Engaging Interceptors
+##### 8.1.3.1 Service Level
+Interceptors could get engaged at service level. One reason for this is that users may 
+want to engage two different interceptor chains for each service even though it is attached to the same Listener. At the 
+listener level resource function paths are relative to the service base path.
+```ballerina
+@http:ServiceConfig{
+   interceptors: [requestInterceptor, responseInterceptor]
+}
+```
+
+##### 8.1.3.2 Listener Level
+Interceptors could get engaged at Listener level as well. At the listener level resource function paths are 
+relative to the /.
+```ballerina
+listener http:Listener echoListener = new http:Listener(9090, config = {interceptors: [requestInterceptor, responseInterceptor]});
+```
+
+##### 8.1.3.3 Execution Order of Interceptors
+These interceptors are always there in the pipeline and executed as the last.
+
+![img.png](img.png)
+In the above example blue dashed box represents the RequestErrorInterceptor and blue boxes simply represent the 
+RequestInterceptors whereas green dashed box represents the ResponseErrorInterceptor(TBA) and green boxes simply represent the 
+ResponseInterceptors. The new execution orders is as follows,
+```
+RequestInterceptor/RequestErrorInterceptor: 1, 2, 4, 5
+ResponseInterceptor/ResponseErrorInterceptor: 5, 3, 0
+```
+
```

Comment:
Do we need to say about the execution order when interceptors are engaged at both service and listener levels?

`ListenerLevel : RequestInterceptors` -> `ServiceLevel : RequestInterceptors` -> `TargetService` -> `ServiceLevel : ResponseInterceptors` -> `ListenerLevel : ResponseInterceptors`